p2pserver/message/types/emergency_req.go

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2018 The ontology Authors
+ * This file is part of The ontology library.
+ *
+ * The ontology is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ontology is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with The ontology.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package types
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"io"
+
+	"github.com/ontio/ontology-crypto/keypair"
+	"github.com/ontio/ontology/common"
+	"github.com/ontio/ontology/common/serialization"
+	"github.com/ontio/ontology/core/types"
+	pc "github.com/ontio/ontology/p2pserver/common"
+)
+
+type EmergencyReason uint8
+
+const (
+	FalseConsensus EmergencyReason = iota
+	StoppedConsensus
+)
+
+type EmergencyEvidence uint8
+
+const (
+	ConsensusMessage EmergencyEvidence = iota
+	ConflictBlock
+)
+
+type EmergencyActionRequest struct {
+	Reason           EmergencyReason
+	Evidence         EmergencyEvidence
+	ProposalBlkNum   uint32
+	ProposalBlk      *types.Block
+	ProposerPK       keypair.PublicKey
+	ProposerSigOnBlk []byte
+	AdminSigs        []*types.Sig
+	ReqPK            keypair.PublicKey
+	ReqSig           []byte
+	hash             *common.Uint256
+}
+
+// Serialize the EmergencyActionRequest
+func (this *EmergencyActionRequest) Serialize(w io.Writer) error {
+	var err error
+	_, err = w.Write([]byte{byte(this.Reason), byte(this.Evidence)})
```

Comment:
why not serialization.WriteUint8 ?

Reply:
WriteUint8 also calls writer.Write(), no difference.
func WriteUint8(writer io.Writer, val uint8) error {
	var p [1]byte
	p[0] = byte(val)
	_, err := writer.Write(p[:])
	return err
}

Reply:
if no difference, why not serialization.WriteUint8 ?

Reply:
I mean use Uint8, twice type conversion: Enum --> uint8-->byte, and it's ok to use enum-->byte directly.